Improvised Weapons Update: Part 1

KathrinBailey opened the pull request:
## About The Pull Request
Upon (more than fair) feedback by Kevin, this PR is being atomised into smaller parts.
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/12170

This is part 1.

• Adds new sprites for the improvised rifle and improvised shotgun.
• Adds a new weapon crafting menu with new parts.
• ~~Adds 20 gauge ammo type.~~ - Changed to use a damage modifer.
• ~~Makes new ammo printable in the autolathe.~~ - Changed to use a damage modifier.
• Replaces instances of modular receivers in the code.

## Why It's Good For The Game
Even though this is just part 1, this actually makes the improvised rifle worth having in the game. Beforehand, it was worse than the shotgun in every conceivable way possible. It did more damage, can be dual-wielded, was cheaper to craft, can be sawn off to fit in backpacks and it was faster to fire. Now the rifle is actually worth considering, even though it is still massively inferior to the shotgun.
Part 2 will add in more variety to improvised weapons.
Also, you shouldn't be able to put a SecOff in their armour vest into crit in a single fucking click with only assistant gear that you can get in about under two minutes. You still can, but it's now harder than it was before.

## Changelog
:cl:
add: New weapon crafting parts for improvised weapons (will be utilised more in future PRs)
balance: Improvised shotguns now have a 0.8 damage modifier, credit ro Kevin for doing this.
balance: Improvised rifles and shotguns require more tools to craft and more parts to craft.
tweak: Descriptions of improvised weapons a bit.
tweak: New sprites for improvised weapons.
/:cl:

## Future Plans
Alright, as mentioned a bit too much, this is part 1. What's coming up:
Part 2
• Improvised Pistol using 32 ACP. You craft your own magazine and get ammo from autolathes.
• Improvised Laser.
• Crafting parts for the laser.
• Loot table for improvised parts
Part 3
• Sumpter rifle. Will be rebalanced to use 32 ACP and be an alternative to the WT550. It'll have its own cargo crate.
Part 4
• Type 21 EED - It's functionally identical to the HoS laser.
• Type 20 EED - It's functionally identical to a laser gun.
• Type 19 EED - Admin-spawn, for events. Suppressive weapon that fires in bursts of three. It's more scary than it is damaging to the crew. It does 6 burn 5 stamina damage for each shot. Connecting all three shots does 2 less damage than the Type 20. Seeing three laser beams coming down the corridor is scary though.

## Acknowledgements
• Necromanceranne  / SpookyNecromancer#24 for going over this and reviewing it, very big help.
• Kevinz for making making a projectile damage modifier that made this PR not require a second ammo type for just the improvised shotgun. Bonus thanks for reviewing this in the first place.
• Lilly for giving this a look over too.
Thanks you three! <3

KathrinBailey committed:
Improvised Weapons Part 1: Shotgun & Rifle

Adds new sprites for ammo, the improvised rifle, the improvised shotgun. Adds a hand saw. Adds things to the crafting menu. Adds things to the autolathe menu.

KathrinBailey committed:
Merge branch 'master' into Improvised-Part-1

necromanceranne commented on code/modules/projectiles/ammunition/ballistic/shotgun.dm:
```diff
@@ -143,3 +143,32 @@
+	projectile_type = /obj/item/projectile/bullet/shotgun_slug/twentybore_slug
+	custom_materials = list(/datum/material/iron=3000)
+
+/obj/item/ammo_casing/shotgun_20bore/buckshot
```
You need to add the pellet count to this ammo type. Check buckshot. Otherwise it only shoots one pellet.

Also the variance or else it won't spread.

KathrinBailey replied:
I think originally this was a child of normal buckshot, so it'd inherit the code for firing 6 pellets and the variance. I'll make sure to double check this, thanks a ton.

necromanceranne commented on code/modules/projectiles/boxes_magazines/internal/shotgun.dm:
```diff
@@ -44,7 +44,9 @@
 /obj/item/ammo_box/magazine/internal/shot/improvised
 	name = "improvised shotgun internal magazine"
 	ammo_type = /obj/item/ammo_casing/shotgun/improvised
+	ammo_type = /obj/item/ammo_casing/shotgun_20bore // New improvised shotgun fires 20 gauge instead of 12 gauge.
```
Why does this have two of the same var? This not the way to do this. Either it needs to be using a list or you need to make improvised 20 bore shells.

Have a look at this pr as an example of how you might implement that: https://github.com/Citadel-Station-13/Citadel-Station-13/pull/10641

KathrinBailey replied:
Oops, a mistake from copying the code to atomise the PR into multiple smaller ones, thanks for pointing it out.

necromanceranne commented on code/modules/research/designs/autolathe_desings/autolathe_designs_sec_and_hacked.dm:
```diff
@@ -73,46 +84,88 @@
 
+/*
 /datum/design/receiver
 	name = "Modular Receiver"
```
Just delete it.

silicons replied:
delete this

necromanceranne commented on code/modules/projectiles/projectile/bullets/shotgun.dm:
```diff
@@ -112,3 +112,24 @@
+
+/obj/item/projectile/bullet/incendiary/shotgun/twentybore_incendiary
+	name = "incendiary slug"
+	damage = 16
```
Just as a slight heads up from experience. Anyone using incendiaries is aiming to set people on fire, not necessarily looking for direct bullet damage. Both this and the 12g will hit for the same fire stacks (4). Which makes them pretty much indistinguishable for the function you want from them.

KathrinBailey replied:
I kinda just wanted to lower the bullet damage because it's a smaller shell. Sadly I ain't American so I have no idea if the hot flash from however much powder is in the 12G round is going to set people on fire more than however much powder is in the 20G round, sadly I can't test it and find out.
It is admittedly more or less the same as the 12G incendiary one which does 20 damage (I think) and the 4 firestacks, I could lower the direct damage by a lot and make it rely more or less solely on the firestacks to deliver damage? I guess the real damage that comes from this isn't as much the burn damage from being set on fire, I guess it's vulnerability you have to put yourself in to put it out, unless you can find an extinguisher.

necromanceranne replied:
Don't worry about realism for the same of accuracy to anything in reality. Think about this from a gameplay perspective. If you want to differentiate the two, it's not a bad move to weaken the bullets firestacks.

KathrinBailey replied:
Marking as resolved since it's being removed.
(Seriously though, thanks for all your time going over this.)

necromanceranne commented on code/modules/projectiles/projectile/bullets/shotgun.dm:
```diff
@@ -112,3 +112,24 @@
+
+/obj/item/projectile/bullet/pellet/shotgun_buckshot/twentybore_buckshot
+	name = "20g buckshot pellet"
+	damage = 10
```
If we assume that you give these the same pellets as buckshot (and since as of writing this you haven't so I couldn't say), then you may wish to be aware this is still 60 damage. Against a security officer that's reduced to 6 per pellet. 42 total. Compared to buckshot (12.5 to 8.75, 52.5 total) that's still not at all bad for hitting into seccies. It slows them at the very least.

KathrinBailey replied:
I think it should inherit whatever code that decides the 12g buckshot fires 6 pellets but instead just use a different pellet. I'll double check this. Going to wait on Kevin first though in case he decides to close it.
Vests have 0.3 bullet armour so it'll be 7 by default, the jumpsuit only has melee protection (I assume you meant 7 anyway since you said 42 total).
The idea behind it is that because the firing action is faster with the shotgun (fire + tap Z, load a new shell vs z, fire, z, load, z, fire, z, load, z fire) and you get a more diverse range of ammo, the shotgun should do a bit less damage. Making the shotguns use 20G and doing less damage is the only real way to make it not a straight upgrade to the rifle.

necromanceranne replied:
Don't have it inheriting from 12g. Have it inherit from a parent shotgun shell unless you want to have guns using the parent ammo using these as well. Would it be realistic for the double-barrel to fit these new shells, for example, or for a bulldog magazine to fit these?

Edit: wait sorry this is for the projectile. But if you change the shell casing, with is what is handled in ammo_casing, keep the above in mind.

KathrinBailey replied:
The caliber in the ammo casing decides what you can load into a magazine. If I try to load casings with "shotgun20bore" it _should_ reject it from Bulldogs and Double-Barrels.
I'll make sure to test it all, I think I can make it a child/subtype of 12G and change the calibre, so there's no issues of it being loaded into the wrong guns.
In terms of double-barrel shotguns or bulldogs fitting 20G:
You'd need a 12G round that essentially has another bullet in it to so there's no chances of causing misfires. A misfire will probably just break the inside of the gun and make it inoperable without taking it apart and doing a bunch of repairs.
You could, if you wanted to, make either the double-barrel or bulldog only fire 20G. It's probably be a way to nerf the Bulldog.

Just a note on that though, there's a limited variety of 20G rounds you can get, At the time I just wanted to limit what you could have because, well, its improvised. In retrospect I probably should've made more 20G rounds.

necromanceranne commented on code/modules/projectiles/projectile/bullets/shotgun.dm:
```diff
@@ -112,3 +112,24 @@
+
+/obj/item/projectile/bullet/shotgun_slug/twentybore_slug
+	name = "12g shotgun slug"
+	damage = 50
```
As the above. 30% of 50 is still only 35. Comparing to 12g slugs (60 to 42), still not bad at all.more importantly I can wipe out their limbs with this still, where their armor isn't.

KathrinBailey replied:
I've looked at this like ten times and it has now only clicked that this is called "12g shotgun slug", I'm an idiot.
It's not that much damage for a shotgun, but it's the only way I can make the shotgun an alternative to the improvised rifle instead of a direct upgrade in every way. With the 12G slugs doing 60 brute, the same as the rifle bullet, the rifle is just kinda left in the cold because it can't be sawn off and its firing action is much slower than the shotgun's firing action.
Initially I thought I could just make the shotgun more expensive to craft, but the thing is, the only thing this achieves is that instead of taking three minutes to get a shotgun, it takes four minutes.
I think it might have been Justin who initially suggested different ammo types? I can't remember, heck.

(Thanks for pointing out I misnamed the slug though heck that's an embarrassing msitake.)

necromanceranne commented on code/modules/research/designs/autolathe_desings/autolathe_designs_sec_and_hacked.dm:
```diff
@@ -73,46 +84,88 @@
 /datum/design/shotgun_slug
 	name = "Shotgun Slug"
-	id = "shotgun_slug"
+	id = "12g shotgun_slug"
```
I'll be honest, I'd keep these to seclathe now. If the bartender wants lethal 12g he should be asking security.

KathrinBailey replied:
These were already in the hacked autolathe.
Like a derp I changed the ID instead of the name though, I meant to change the name to 12g so it'd be a bit more consistent in the menu.
Removing 12g lethals from the hacked autolathe is something Kevinz would have to weigh in on.

necromanceranne commented on code/datums/components/crafting/recipes/recipes_weapon_and_ammo.dm:
```diff
@@ -394,3 +398,60 @@
+	result = /obj/item/weaponcrafting/improvised_parts/shotgun_receiver
+	reqs = list(/obj/item/stack/sheet/metal = 10,
+				/obj/item/stack/sheet/plasteel = 1)
+	tools = list(TOOL_SCREWDRIVER, TOOL_WELDER) // Increased cost is to stop dual-wield alpha striking. ishotgun is a rvolver and can be duel-wielded
```
Weird spot for this code comment but have you considered locating where ever in gods name the check to stop shotguns from being dual-wielded is and replacing it with a more generic flag?

KathrinBailey replied:
I originally wanted to make them two-handed so they have an actual reasonable position in the game instead of being the the unbalanced as fuck alpha strike for a literal free crit.
Problem is, people will _lose their shit_ if I tried to make the shotgun harder to craft, slightly less damage **and** made them two-handed.
Fortunately, another PR has gone up to make them two-handed, so that should address this.
I'm glad someone else sees the sense in making them two-handed since this PR really doesn't really bring the ishotgun in line as much as it needs to since the alpha strike is still enough to crit.
Once I get a merge conflict for the PR to make ishotguns two-handed I'll sigh in relief.

KathrinBailey replied:
I wanted to make the shotguns two-handed at first, but I was worried that this would get the PR too much backlash. Even though that's the correct balance decision to make (the rifle is twohanded), I'm already nerfing the damage by locking it to inferior ammo and I'm making it more expensive to craft.
Fortunately though, someone else made a PR to make the ishotgun two-handed. Hopefully that gets merged so it'll become two-handed. I suppose once I see a conflict I know that the two-handed PR got merged so I can remove the comment.
I'm glad someone else thinks this should be two-handed though.

necromanceranne replied:
I don't think anyone is going to yell at you for consistency. It's just I feel like the heavy weapon change fits the bigger shotguns due to using a pump action, while these not needing an empty hand is more, to put it in the best terms, 'cinematic'. From a pure gameplay standpoint there is good reason to not have them dual-wieldable. But the fix isn't making them heavy and you can save me and everyone some time having to make that change by making it yourself if you can figure it out.

Ask kev, he may know a solution. Dual-wielding was his code. @kevinz000

KathrinBailey replied:
Just to throw the idea out there, maybe making the full length weapon two-handed, but making the sawn-off one-handed but there's some sort of disadvantage using it without a free-hand or alternatively a disadvantage to sawing it off in the first place could work?
At the moment, there's no reason _not_ to saw it off.

KathrinBailey replied:
Ghommie's PR should solve the two-handed issue I think.

YakumoChen commented on code/datums/components/crafting/guncrafting.dm:
```diff
@@ -19,3 +22,49 @@
+
+/obj/item/weaponcrafting/improvised_parts
+	name = "Nuclear Syndicate Hyper Giga-Operative Undercover 5th-Dimensional Super-Tank"
+	desc = "An undercover hypno sissification operative attempting to make Nanotrasen homosexual. Report this to Central immediately. Something is off here."
```
Please Don't do this. Just give it a normal name.

YakumoChen replied:
Also you do not need to add your name to everything you code in the code. I've seen this regularly from you in PRs and you should knock it off.

YakumoChen replied:
This is not resolved. Change it to a normal debug name and description.

silicons commented on code/datums/components/crafting/guncrafting.dm:
```diff
@@ -19,3 +22,49 @@
+
+/obj/item/weaponcrafting/improvised_parts
+	name = "Nuclear Syndicate Hyper Giga-Operative Undercover 5th-Dimensional Super-Tank"
+	desc = "An undercover hypno sissification operative attempting to make Nanotrasen homosexual. Report this to Central immediately. Something is off here."
```
However, let's not, silly code comments aside. C'mon.

silicons reviewed (changes requested):
Right this is waiting on 
#12274
and
#12293

Trilbyspaceclone commented on code/modules/research/designs/autolathe_desings/autolathe_designs_sec_and_hacked.dm:
```diff
@@ -7,20 +7,31 @@
+	name = "12g Beanbag Slug"
 	id = "beanbag_slug"
 	build_type = AUTOLATHE
 	materials = list(/datum/material/iron = 250)
```
This is double defined

silicons reviewed (changes requested):
Right, update:

Do not make improvised shotgun require snowflake ammo, instead give it a 0.8x damage multiplier with regular ammo.
Also please replace the hypno or whatever desc on the base path item to something less silly.

Everything else looks good @KathrinBailey

silicons commented on code/datums/components/crafting/guncrafting.dm:
```diff
@@ -1,13 +1,16 @@
 /obj/item/weaponcrafting/receiver
 	name = "modular receiver"
 	desc = "A prototype modular receiver and trigger assembly for a firearm."
+	name = "broken gun part" // Replaced with new receivers below!!
```
double defined

silicons commented on code/datums/components/crafting/guncrafting.dm:
```diff
@@ -1,13 +1,16 @@
 	name = "modular receiver"
 	desc = "A prototype modular receiver and trigger assembly for a firearm."
+	name = "broken gun part" // Replaced with new receivers below!!
+	desc = "An irreparably broken receiver. This is beyond saving and cannot be used. OOC: Outdated item, report to coders, you shouldn't get this." // Replaced with new receivers below!!
```
just remove this if it's unused, why is this being kept?

silicons commented on code/modules/research/designs/autolathe_desings/autolathe_designs_sec_and_hacked.dm:
```diff
@@ -73,46 +84,88 @@
+	name = "Rifle Receiver"
+	id = "rifle_receiver"
+	build_type = AUTOLATHE
+	materials = list(/datum/material/iron = 40000)
```
modify both this and the recipe to take 20 sheets of iron, reminder that I've nerfed material upgrade amounts, no need to balance by 0's anymore.

silicons commented on code/modules/research/designs/autolathe_desings/autolathe_designs_tcomms_and_misc.dm:
```diff
@@ -271,4 +271,13 @@
+	name = "Trigger Assembly"
+	id = "trigger_assembly"
+	build_type = AUTOLATHE
+	materials = list(/datum/material/iron = 10500, /datum/material/glass = 50)
```
Um. 5 sheets metal? 6500 (around 3.25 sheets) would be a bit more reasonable for a *trigger*.

KathrinBailey replied:
Changed to this, the 0.25 iron sheets and glass is from the igniter that you'd use to otherwise craft it manually.

silicons commented on code/game/objects/items/tools/saw.dm:
```diff
@@ -0,0 +1,47 @@
+		var/our_color = pick(saw_colors)
+		add_atom_colour(saw_colors[our_color], FIXED_COLOUR_PRIORITY)
+		update_icon()
+	if(prob(75))
```
Er...?

silicons commented on code/game/objects/items/tools/saw.dm:
```diff
@@ -0,0 +1,47 @@
+		add_atom_colour(saw_colors[our_color], FIXED_COLOUR_PRIORITY)
+		update_icon()
+	if(prob(75))
+		pixel_y = rand(0, 16)
```
-8, 8, unless you specifically want this skewed upwards.

KathrinBailey committed:
PR not yet ready - just cleaning issues.

Just trying to remark some of the issues resolved so I can see where I'm at a bit easier.

KathrinBailey committed:
Reverting changes to shotgun ammo names in the autolathe.

No point calling it 12G when there's no 20G anymore.

KathrinBailey committed:
Merge branch 'master' into Improvised-Part-1

KathrinBailey committed:
Weapon weight + sprite improvements.

Makes sprites for gun parts bulkier with a low alpha pixel outline to make them easier to click on. They also look a bit closer to what their final product will be.

KathrinBailey committed:
Merge branch 'Improvised-Part-1' of https://github.com/KathrinBailey/Citadel-Station-13 into Improvised-Part-1

KathrinBailey committed:
oops forgot the "

KathrinBailey committed:
Revert "Merge branch 'master' into Improvised-Part-1"

This reverts commit 1e43d228d1bc35b0582244ed0bd91934573b3cdd, reversing
changes made to 207e9fccd4e8cffbe9ed26401a6f516fc44245dd.

KathrinBailey committed:
Revert "Revert "Merge branch 'master' into Improvised-Part-1""

This reverts commit 22d421d010c4a4c5c5e26f49fe6bffcf6198df42.

KathrinBailey committed:
Merge remote-tracking branch 'upstream/master' into Improvised-Part-1

KathrinBailey committed:
My Github or environment isn't flipping out. no more.

Derp. Also projectile damage modifier fixed. I can actually compile without hundreds of errors.

KathrinBailey committed:
Made small weapon parts have w_class small.

I was testing it out and seeing the trigger assembly take so much space in my inventory seemed really odd.

KathrinBailey committed:
Duplicate definition refused.

Hurrdurr I suck.
Glad I gave this a quick look-voer.
Shotgun description edited to reflect the 0.8x damage modifier.

KathrinBailey committed:
Resolved tools.dmi merge conflict.

Just took the dmi from master and put the new sprites on at the end.

KathrinBailey committed:
DNM - Trying to see why usptream tools.dmi with my stuff added on the end is conflicting.

KathrinBailey committed:
Please don't conflict

KathrinBailey committed:
Merge remote-tracking branch 'upstream/master' into Improvised-Part-1
